scripts/builtin/imputeByKNN.dml

```diff
@@ -0,0 +1,208 @@
+#-------------------------------------------------------------
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+#-------------------------------------------------------------
+
+# Impute the data by using KNN-algorithm and finding the nearest neighbor using euclidean distance
+# Currently only for single column with multiple missing values and top 1 neighbor
+#
+# INPUT:
+# -------------------------------------------------------------------------------------
+# data    Data Matrix (numerical)
+# method  methods of calculating the KNN-algorith depending on the size of data and missing values
+# -------------------------------------------------------------------------------------
+#
+# OUTPUT:
+# -----------------------------------------------------------------------------------
+# result     imputed dataset
+# -----------------------------------------------------------------------------------
+
+
+#method 1
+#impute By Mean, really small, coarse grained-operation
+#Extract Top 1 distances, impute the respective value
+#dist runtime * #features
+#method 2
+#assuming missing value is very small, 1%
+#compute only the distances between tuples between missing value, impute the mean,
+#compute distance between entire "X" (potentially large) and rows with missing values (hopefully small),
+#method 3
+#sample rows, randomly select subset of rows
+#assuming missing value is very large,
+#compute distance between sample of rows of X (control small) and rows with missing values (if not small),
+
+m_imputeByKNN = function(Matrix[Double] data, String method = "default")
+return(Matrix[Double] result)
+{
+  print("KNN-IMPUTATION SCRIPT")
+
+  #Test data, initial data, will be replaced later with variable data
+  first = matrix ("1 3 4 8", rows = 4, cols = 1)
+  second = matrix ("2 4 6 7", rows = 4, cols = 1)
+  third = matrix ("4 NaN 5 NaN", rows = 4, cols = 1)
+  test_matrix = cbind(first,second,third)
```

Comment:
This test data cannot be in the builtin function, which is supposed to be general purpose. Move this to the tests.